Sources/NIOHTTPClient/HTTPHandler.swift

```diff
@@ -484,6 +476,6 @@ internal struct RedirectHandler<T> {
             request.headers.remove(name: "Proxy-Authorization")
         }
 
-        return self.execute(request).cascade(promise: promise)
+        return self.execute(request).future.cascade(to: promise)
```

Comment:
Originally, I wanted `Task` to be sort of like a future, with all methods of the future. I think those methods make using `Task` a bit simpler.